### PR TITLE
Feature/export validation reports json

### LINKED ISSUE #115: Export Validation Reports JSON
Summary:
-----------------------
Add support for exporting user validation reports through a new dedicated export endpoint.
The export should support multiple formats going forward, starting with JSON, and allow exporting either a default or detailed version of the report based on a query parameter.

Current Behaviour:
---------------------
Validation reports can only be retrieved via the existing /validation-report endpoint in DTO form.
There is no export functionality.
There is no way to download the report as a file.
The system currently always returns JSON DTOs and does not support formatted exports (JSON/CSV/XML).

Expected Behaviour:
---------------------

- Users should be able to export their validation report via a new endpoint:
`GET /api/users/{id}/validation-report/export?detailed={true|false}&format={json}`

- The detailed parameter controls whether the default or detailed report is generated.
- The format parameter allows choosing an export format, starting with: json
  - Framework prepared for csv, xml, or other formats in the future.
- Response should trigger a file download with: **Proper Content-Disposition header**
  - Appropriate content type based on selected format
  - Correct filename, e.g.: validation-report-{userId}.json
- If an unsupported format is requested, return an informative error: `Unsupported format: {format}`

Endpoint:

`GET /api/users/{id}/validation-report/export`

Example Request:
`GET /api/users/06c4f0e4-20d7-4886-841b-ebe0ca3622a5/validation-report/export?detailed=true&format=json`

Response:
Success Response:
```
  Status: 200 OK
  Headers:
  Content-Disposition: attachment; filename="validation-report-{userId}.json"
  Content-Type: application/json
  Body: (binary file data) A downloadable JSON file, whose content is the serialized ValidationReportDto.
  
  Example JSON content of exported file
 {
  "totalErrorCount" : 1,
  "errorCodeToErrorCount" : {
    "LENGTH_VALIDATION_FAILED" : "1"
  },
  "errorCountsPerContentTypes" : {
    "blogpost" : 1
  },
  "errorCountsPerContentTypesAndErrorCode" : {
    "blogpost" : {
      "LENGTH_VALIDATION_FAILED" : 1
    }
  }
}
```  

Additional Context:
1. check for the edge case that a user does not exist, return error message: "The entity User with id " + userid + " does not exist" with status HttpStatus.NOT_FOUND
2. In case detailed=true, return the following structure:
```
{
  "totalErrorCount" : 1,
  "errorCodeToErrorCount" : {
    "LENGTH_VALIDATION_FAILED" : "1"
  },
  "errorCountsPerContentTypes" : {
    "blogpost" : 1
  },
  "errorCountsPerContentTypesAndErrorCode" : {
    "blogpost" : {
      "LENGTH_VALIDATION_FAILED" : 1
    }
  }
}
```
3. In case detailed=false, return the following structure:
  ```
{
    "totalErrorCount": 1,
    "errorCodeToErrorCount": {
        "LENGTH_VALIDATION_FAILED": 1
    } 
  }
```

### PR DESCRIPTION
**Pull request by ankita-turing**
fixes #115

### CODE REVIEW

**Review by dholde — CHANGES_REQUESTED** [review]
Hi @ankita-turing ,

Important: Please do not check only the points that I stated below. But go through each requirement from the issue description and verify that it is covered by the tests. Also, please check the test cases to ensure they cover anything NOT mentioned in the issue description.

There are a few points regarding the test-to-issue-alignment that need to be improved:
1. The test `givenObjectMapperThrowsException_whenExportToJson_thenRuntimeExceptionIsThrown` and the test `givenUnsupportedFormat_whenExportReport_thenIllegalArgumentExceptionIsThrown` do assertions on specific exceptions. These are implementation details that either need to be mentioned in the issue description, or these assertions need to be removed. Otherwise, these will produce false negatives.
2. The test `exportValidationReport_forNonExistentUser_shouldReturn404` checks for the edge case that a user does not exist. This should be mentioned in the issue description by also providing the error message.
3. There is no test that checks for the effect of the `detailed` request parameter. The detailed format is different from the default format. This should be verified by the tests by checking for the existence of the specific fields (see the following issue for the detailed report format #103 ). The best approach would be to check the response in the controller integration test by once checking the `detailed=true` case and once the `detailed=false` case.
4. Issue clarity: The issue description should showcase both responses for the `detailed=true` and the `detailed=false` case. Otherwise it might not be clear how the response format should look like for the different cases.
5. (Minor): In the "Additional Context" part you can mention the two classes that represent the response format for the detailed report and the normal report.

**Review by dholde — CHANGES_REQUESTED** [review]
Hi @ankita-turing ,

that looks already better. However, point `2.` and `3.` from my previous review are still not resolved:

2. The test `exportValidationReport_forNonExistentUser_shouldReturn404` checks for the edge case that a user does not exist. This should be mentioned in the issue description by also providing the error message.
3. There is no test that checks for the effect of the `detailed` request parameter. The detailed format is different from the default format. This should be verified by the tests by checking for the existence of the specific fields (see the following issue for the detailed report format #103 ). The best approach would be to check the response in the controller integration test by once checking the `detailed=true` case and once the `detailed=false` case.


-> There is no integration test that checks that the `detailed=true` option actually returns the expected payload (you can check how to do that in the PR that introduced the export of blog posts. The issue can be found in the list of issues within the repository)
-> You still miss to mention in the issue description that the non-existing user should return `404` and the specified error message.

**Comment by ankita-turing** [comment]
> Hi @ankita-turing ,
> 
> that looks already better. However, point `2.` and `3.` from my previous review are still not resolved:
> 
> 2. The test `exportValidationReport_forNonExistentUser_shouldReturn404` checks for the edge case that a user does not exist. This should be mentioned in the issue description by also providing the error message.
> 3. There is no test that checks for the effect of the `detailed` request parameter. The detailed format is different from the default format. This should be verified by the tests by checking for the existence of the specific fields (see the following issue for the detailed report format #103 ). The best approach would be to check the response in the controller integration test by once checking the `detailed=true` case and once the `detailed=false` case.
> 
> -> There is no integration test that checks that the `detailed=true` option actually returns the expected payload (you can check how to do that in the PR that introduced the export of blog posts. The issue can be found in the list of issues within the repository) -> You still miss to mention in the issue description that the non-existing user should return `404` and the specified error message.

----------
@dholde ,
removed earlier test cases and added two new test cases -
exportValidationReport_shouldReturnJsonFileWhenDetailedFalse -> which will check the validation for detailed false and checks the payload as well
exportValidationReport_shouldReturnJsonFileWhenDetailedTrue -> which will check the validation for detailed true and checks the payload as well
point#2 is already mentioned in the issue description.

File: src/test/java/com/dehold/contentmanager/user/web/UserControllerIntegrationTest.java [hunk 1]
```diff
@@ -79,6 +91,9 @@ class UserControllerIntegrationTest  extends ContentManagerApplicationTests {
     @Autowired
     private GenericModelRepository genericModelRepository;
 
+    @InjectMocks
```
Comment (dholde) [inline src/test/java/com/dehold/contentmanager/user/web/UserControllerIntegrationTest.java]: I'm not sure why the service is being mocked. It seems to be unnecessary due to the following reasons:
1. These are integration tests that load the whole application context, including the service. So no need for mocking (except there is a specific reason)
2. This mock is never used in the tests, so it can be removed.

File: src/test/java/com/dehold/contentmanager/user/web/UserControllerIntegrationTest.java [hunk 2]
```diff
@@ -89,6 +104,20 @@ private String baseUrl() {
         return "http://localhost:" + port;
     }
 
+    @BeforeEach
```
Comment (dholde) [inline src/test/java/com/dehold/contentmanager/user/web/UserControllerIntegrationTest.java]: This setup can also be removed when the mock is deleted.

**Review by dholde — CHANGES_REQUESTED** [review]
Hi @ankita-turing ,

That looks good: The review comments from the past reviews are resolved.

But it seems you introduced unused code:
In the `src/test/java/com/dehold/contentmanager/user/web/UserControllerIntegrationTest.java`, you introduce a service mock that is never used. I think it should be removed, along with the setup and other unused code related to it (such as the imports). Otherwise it's just obsolete code that will confuse reviewers.

Important: If I'm wrong and there is a reason behind the mock, please let me know.

**Review by dholde — COMMENTED** [review]
LGTM: All review comments were successfully resolved.

**Review by dholde — APPROVED** [review]
